packages/graphql-hooks/src/types/common-types.ts

```diff
@@ -155,10 +180,13 @@ export interface HttpError {
 export interface APIError<TGraphQLError = object> {
   fetchError?: Error
   httpError?: HttpError
-  graphQLErrors?: TGraphQLError[]
+  graphQLErrors?: GraphQLResponseError[]
 }
 
-export interface Result<ResponseData = any, TGraphQLError = object> {
+export interface Result<
+  ResponseData = any,
+  TGraphQLError = GraphQLResponseError
+> {
```

Comment:
Wasn't too sure how updating these generics would effect existing typings so I opted to let the default be the spec defined error, just incase an override was ever useful.